src/main/java/app/model/agents/Grid.java

```diff
@@ -49,15 +49,31 @@ public Grid(double length, double width, CellType type, double cellSize)
 
     public Cell getCellAt(int row, int col)
     {
-        return grid[row][col];
+        try
+        {
+            Cell cell = grid[row][col];
+            return cell;
+        }
+        catch(Exception e)
+        {
+            System.out.println("Cell index out of bounds");
```

Comment:
This exception isn't actually handled?  or is returning null the desired action of the catch?

Reply:
Its the desired action, as it keeps track with a set array of sensing directions